evm/src/fixed_recursive_verifier.rs

```diff
@@ -655,6 +655,11 @@ where
         lhs: &ExtraBlockDataTarget,
         rhs: &ExtraBlockDataTarget,
     ) {
+        // Connect genesis state root values.
+        for (&limb0, &limb1) in pvs.genesis_state_root.iter().zip(&rhs.genesis_state_root) {
+            builder.connect(limb0, limb1);
+        }
```

Comment:
Don't we need to also connect `lhs.genesis_state_root` and `pvs.genesis_state_root`?